server/src/main/java/io/deephaven/server/barrage/BarrageStreamGenerator.java

```diff
@@ -532,12 +546,16 @@ private long appendAddColumns(final View view,
                 try (WritableRowSet intersect = view.keyspaceViewport().intersect(rowsIncluded.original)) {
                     myAddedOffsets = rowsIncluded.original.invert(intersect);
                 }
-            } else if (!rowsAdded.original.equals(rowsIncluded.original)) {
-                // there are scoped rows included in the chunks that need to be removed
-                myAddedOffsets = rowsIncluded.original.invert(rowsAdded.original);
             } else {
-                // use chunk data as-is
-                myAddedOffsets = null;
+                if (view.isInitialSnapshot()) {
+                    // use chunk data as-is, rely on snapshot process to avoid duplicates
```

Comment:
`rowsAdded` is decoupled from `rowsIncluded` in growing viewport subscription snapshot messages.  Since viewport subscriptions are given priority, full subscriptions may see rows snapshotted that were previously provided to the client.  The code treats `snapshot` messages separately from `delta` messages.